cloud_Azure/terraform/module/network_watcher.tf

```diff
@@ -1,18 +1,19 @@
-# Network Watcher is created automatically by Azure when VirtualNetwork is created or updated in the subscription, see:
-# https://docs.microsoft.com/en-us/azure/network-watcher/network-watcher-create
+# Network Watcher is created automatically by Azure when VirtualNetwork is created or updated in the subscription
+# see: https://docs.microsoft.com/en-us/azure/network-watcher/network-watcher-create
 data "azurerm_network_watcher" "network_watcher" {
-  name                = "NetworkWatcher_${var.location}"
+  name                = "NetworkWatcher_${var.location}" # Azure creates NetworkWatcher named exactly that
   resource_group_name = "NetworkWatcherRG"
 }
 
 # Runs python script to gather network security groups from each requested resource group
+# This is required because no Terraform provider exposes such functionality
 # Resulting "data.external.nsg_data_source.results" is a map of string -> string, eg.
 # {
 #   "ResourceGroupName1" -> "NetworkSercurityGroupId1,NetworkSecurityGroupId2",
 #   "ResourceGroupName2" -> "NetworkSercurityGroupId3,NetworkSecurityGroupId4"
 # }
 data "external" "nsg_data_source" {
-  program = ["python3", "${path.module}/get_nsg.py"]
+  program = ["python", "${path.module}/get_nsg.py"]
```

Comment:
Note that on some systems (e.g. macOS up to the most recent version), `python` is still version 2.

Reply:
Hmm, another proof that macOS should be banned... ;)

Reply:
BTW, `python` won't be distributed with macOS starting in version (12.3) ... so it will be more like a typical minimal Linux distro :).